api/v1alpha3/machine_webhook.go

```diff
@@ -56,8 +69,21 @@ func (m *Machine) validate() error {
 			allErrs,
 			field.Required(field.NewPath("spec", "bootstrap", "data"), "expected spec.bootstrap.data or spec.bootstrap.configRef to be populated"),
 		)
+	}
 
+	namespaces := make(map[string]struct{})
+	namespaces[m.Namespace] = struct{}{}
+	namespaces[m.Spec.InfrastructureRef.Namespace] = struct{}{}
+	if m.Spec.Bootstrap.ConfigRef != nil {
+		namespaces[m.Spec.Bootstrap.ConfigRef.Namespace] = struct{}{}
 	}
+	if len(namespaces) != 1 {
+		allErrs = append(
+			allErrs,
+			field.Forbidden(field.NewPath("namespace"), "expected namespace, spec.bootstrap.configRef.namespace and spec.infrastructureRef.namespace to match"),
```

Comment:
It might be better to compare the namespaces individually to provide a more targeted error to users rather than grouping them together like this.